### PR TITLE
[patch] bugfix lb-gateway's Insert rpc nil pointer panic

### PR DESCRIPTION
**Pull request by kpango**
Signed-off-by: kpango <kpango@vdaas.org>

### Description:
bugfix, gateway panics when user didn't send request config.

### Environment:

- Go Version: 1.15.7
- Docker Version: 19.03.8
- Kubernetes Version: 1.18.2
- NGT Version: 1.13.1

### Types of changes:

- [x] Bug fix [type/bug]
- [ ] New feature [type/feature]
- [ ] Add tests [type/test]
- [ ] Security related changes [type/security]
- [ ] Add documents [type/documentation]
- [ ] Refactoring [type/refactoring]
- [ ] Update dependencies [type/dependency]
- [ ] Update benchmarks and performances [type/bench]
- [ ] Update CI [type/ci]

### Changes to Core Features:

- [ ] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your core changes, as applicable?
- [ ] Have you successfully ran tests with your changes locally?

### Checklist:

- [ ] I have read the [CONTRIBUTING](https://github.com/vdaas/vald/blob/master/CONTRIBUTING.md) document.
- [ ] I have checked open [Pull Requests](https://github.com/vdaas/vald/pulls) for the similar feature or fixes?
- [ ] I have added tests and benchmarks to cover my changes.
- [ ] I have ensured all new and existing tests passed.
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: pkg/gateway/backup/handler/grpc/handler.go [hunk 1]
```diff
@@ -1251,7 +1267,11 @@ func (s *server) Remove(ctx context.Context, req *payload.Remove_Request) (loc *
 			}
 			return nil, err
 		}
-		req.Config.SkipStrictExistCheck = true
+		if req.GetConfig() != nil {
+			req.Config.SkipStrictExistCheck = true
+		} else {
+			req.Config = &payload.Remove_Config{SkipStrictExistCheck: true}
+		}
 	}
```
Comment (rinx) [inline pkg/gateway/backup/handler/grpc/handler.go:1275]: How about for MultiRemove? Is it not needed?

File: pkg/gateway/lb/handler/grpc/handler.go [hunk 2]
```diff
@@ -1561,7 +1577,11 @@ func (s *server) Remove(ctx context.Context, req *payload.Remove_Request) (locs
...
 		}
-		req.Config.SkipStrictExistCheck = true
+		if req.GetConfig() != nil {
+			req.Config.SkipStrictExistCheck = true
+		} else {
+			req.Config = &payload.Remove_Config{SkipStrictExistCheck: true}
+		}
 	}
 	var mu sync.Mutex
 	locs = &payload.Object_Location{
```
Comment (rinx) [inline pkg/gateway/lb/handler/grpc/handler.go:1587]: same, is there no need to modify MultiRemove?

Comment (kpango) [reply pkg/gateway/lb/handler/grpc/handler.go:1587]: thank you for review, I fixed it.

**Comment by kpango** [comment]
@rinx Thank you for review, I resolved all suggestion.

File: internal/safety/safety.go [hunk 3]
```diff
@@ -37,12 +38,13 @@ func recoverFunc(fn func() error, withPanic bool) func() error {
...
 			if r := recover(); r != nil {
-				log.Warnf("recovered:\t%+v", r)
+				stack := string(debug.Stack())
+				log.Warn("recovered: %+v\nstacktrace:\n%s", r, stack)
 				switch x := r.(type) {
 				case runtime.Error:
 					err = errors.ErrRuntimeError(err, x)
 					if withPanic {
-						log.Error(err, info.Get())
+						log.Errorf("recovered but this thread going to panic because panic reason is runtimer.Error\nerror: %v\ninfo:\n%s\nstacktrace:\n%s", err, info.Get().String(), stack)
```
Comment (rinx) [inline internal/safety/safety.go:47]:
```suggestion
						log.Errorf("recovered but this thread going to panic: the reason is runtimer.Error\nerror: %v\ninfo:\n%s\nstacktrace:\n%s", err, info.Get().String(), stack)

```

Comment (rinx) [reply internal/safety/safety.go:47]: I thought that info.Get() already contains stack traces. is the `stack` different from that?

Comment (kpango) [reply internal/safety/safety.go:47]: I thought this StackTrace was the same, but our trace uses runtime.Caller, and runtime/debug.Stack uses buffered runtime.Stack and gets more information, so this is more informative for StackTrace.

Comment (kpango) [reply internal/safety/safety.go:47]: but runtime.Stack returns too many information, I think we need both information for debug and we should use debug.Stack only panic case.

Comment (rinx) [reply internal/safety/safety.go:47]: ok

File: internal/safety/safety.go [hunk 4]
```diff
@@ -44,7 +44,8 @@ func recoverFunc(fn func() error, withPanic bool) func() error {
 				case runtime.Error:
 					err = errors.ErrRuntimeError(err, x)
 					if withPanic {
-						log.Errorf("recovered but this thread going to panic because panic reason is runtimer.Error\nerror: %v\ninfo:\n%s\nstacktrace:\n%s", err, info.Get().String(), stack)
+						log.Errorf("recovered but this thread going to panic: the reason is runtimer.Error\nerror: %v\ninfo:\n%s\nstacktrace:\n%s", err, info.Get().String(), stack)
```
Comment (rinx) [inline internal/safety/safety.go:47]:
```suggestion
						log.Errorf("recovered but this is thread going to panic: the reason is runtimer.Error\nerror: %v\ninfo:\n%s\nstacktrace:\n%s", err, info.Get().String(), stack)
```
sorry i forgot.

**Comment by rinx** [comment]
Maybe others okay.

**Comment by kpango** [comment]
I'm going to merge & release this PR after all build succeed.

**Pull request merged**